pkg/minishift/hostfolder/hostfolder.go

```diff
@@ -0,0 +1,428 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostfolder
+
```

Comment:
I am wondering whether we need an interface here for the basic operations offered by the commands? How am I going to plug in sshfs support? Take _Add_ for example. Introduce an interface later?